### ISSUE TITLE
Running granular test cases in a broader test suite of YAML files

### ISSUE THREAD

**Issue by kiminoa** [cdent/gabbi#44]
Is there a way to override the build_test functionality and load tests on a more granular basis?  Something similar to `nosetests -m` to modify the test regex pattern.

Better: Can request only a single YAML file be run instead of all YAML files in various `gabbits/`
Best: Can regex on the `tests: -name:` string.

It's one of two remaining sticking points in my gabbi Proof of Concept.

```python
for test_file in glob.iglob(yaml_file_glob):
        if intercept:
            host = str(uuid.uuid4())
        test_yaml = load_yaml(test_file)
        test_name = '%s_%s' % (test_loader_name,
                               os.path.splitext(
                                   os.path.basename(test_file))[0])
        file_suite = test_suite_from_yaml(loader, test_name, test_yaml,
                                          path, host, port, fixture_module,
                                          intercept)
        top_suite.addTest(file_suite)
    return top_suite
```

**Comment by cdent** [comment]
It depends a lot on the testrunner you are using. If you're using `testrepository` and `subunit` as used in the gabbi tests themselves you can provide a pattern to tox that eventually gets passed on down to filter the tests that are run. For example if you clone the gabbi repo and do:

```
$ tox -epy27 -- self
```

that will run just the tests in the `self.yaml` file. You can also choose a specific test

```
tox -epy27 -- self_bogus_method
```

That will run the 'bogus method' test in the self file.

The reason this works is because testr (the short name for testrepository) does discovery to list all the tests and then starts another process to actually run those tests, filtering according to the rules you provide.

`unittest` and similar tools have ways to limit tests when doing discovery but it does it at the level of file name. So, for example, you can:

```
python -m unittest discover -p 'test_intercept*'
```

to get the tests from just the test_intercept file.

So one way to get what you want is to use testr. Another, if you don't like testr, is to have more test_ files, each of which loads a subset of the YAML files and then use the -p thing.

Does any of that get you closer?

If not there might be some other ways worth exploring, but it would be a while before it was released because I want to get 1.0 out the door soon without too many more changes (except for bug fixes).

**Comment by cdent** [comment]
Closing this as I guess it is resolved?
